DataFormats/TestObjects/interface/SchemaEvolutionTestObjects.h

```diff
@@ -0,0 +1,332 @@
+
+    int fSize_ = 3;
+    int* a_;  //[fSize_]
+#else
+    SchemaEvolutionCArrayToStdVector() : a_{0, 0, 0} {}
+    SchemaEvolutionCArrayToStdVector(int x, int y, int z) : a_{x, y, z} {}
+    std::vector<int> a_;
+#endif
+  };
+
+  class SchemaEvolutionVectorToList {
+  public:
+    SchemaEvolutionVectorToList() : a_{0, 0, 0} {}
+    SchemaEvolutionVectorToList(int x, int y, int z) : a_{x, y, z} {}
+#if defined DataFormats_TestObjects_USE_OLD
+    std::vector<int> a_;
+#else
+    std::list<int> a_;
+#endif
+  };
+
+  class SchemaEvolutionMapToUnorderedMap {
+  public:
+    SchemaEvolutionMapToUnorderedMap() {
+      a_.insert({0, 0});
+      a_.insert({1, 0});
+      a_.insert({2, 0});
+    }
+    SchemaEvolutionMapToUnorderedMap(int keyX, int x, int keyY, int y, int keyZ, int z) {
+      a_.insert({keyX, x});
+      a_.insert({keyY, y});
+      a_.insert({keyZ, z});
+    }
+#if defined DataFormats_TestObjects_USE_OLD
+    std::map<int, int> a_;
+#else
+    std::unordered_map<int, int> a_;
+#endif
+  };
+
+  class VectorVectorElement {
+  public:
+    VectorVectorElement();
+    VectorVectorElement(int a,
+                        int b,
+                        SchemaEvolutionChangeOrder const&,
+                        SchemaEvolutionAddMember const&,
+                        SchemaEvolutionRemoveMember const&,
+                        SchemaEvolutionMoveToBase const&,
+                        SchemaEvolutionChangeType const&,
+                        SchemaEvolutionAddBase const&,
+                        SchemaEvolutionPointerToMember const&,
+                        SchemaEvolutionPointerToUniquePtr const&,
+                        SchemaEvolutionCArrayToStdArray const&,
+                        // SchemaEvolutionCArrayToStdVector const&,
+                        SchemaEvolutionVectorToList const&,
+                        SchemaEvolutionMapToUnorderedMap const&);
+#if defined DataFormats_TestObjects_USE_OLD
+    int a_;
+    int b_;
+#else
+    int a_;
+    int b_;
+    int c_ = 0;
+#endif
+    SchemaEvolutionChangeOrder changeOrder_;
+    SchemaEvolutionAddMember addMember_;
+    SchemaEvolutionRemoveMember removeMember_;
+    SchemaEvolutionMoveToBase moveToBase_;
+    SchemaEvolutionChangeType changeType_;
+    SchemaEvolutionAddBase addBase_;
+    SchemaEvolutionPointerToMember pointerToMember_;
+    SchemaEvolutionPointerToUniquePtr pointerToUniquePtr_;
+    SchemaEvolutionCArrayToStdArray cArrayToStdArray_;
+    // This one is commented out because it fails reading an old format
+    // input file with an executable built with the modified format.
+    // If the issue in ROOT is ever fixed and this is added back,
+    // it also would need to be added into the constructor above.
+    // SchemaEvolutionCArrayToStdVector cArrayToStdVector_;
+    SchemaEvolutionVectorToList vectorToList_;
+    SchemaEvolutionMapToUnorderedMap mapToUnorderedMap_;
+  };
+
+  class VectorVectorElementNonSplit {
```

Comment:
What makes this class a "non-split"? Based on the name I was expecting to see something like `splitLevel="0"` in the `classes_def.xml`, but maybe I misunderstood the purpose?

Reply:
The "old" version of VectorVectorElementNonSplit contains only a single integer. It cannot be split because there is only one data member.

Maybe using ```splitLevel = "0"``` would also work. I didn't try. I can try that if you think it is worth the effort. If the class is split the error does not occur. I had originally thought the other classes would also exhibit the problem, but they don't. This is the only one that does. I actually added it at the end of the development when I realized that.

Reply:
> The "old" version of VectorVectorElementNonSplit contains only a single integer. It cannot be split because there is only one data member.

Ok. Could you add a comment about that?

> If the class is split the error does not occur.

What error do you refer to? (what did I miss?)

Reply:
The original bug in ROOT in CMSSW_13_0_0 that started all this only affected non split classes.

Reply:
You might want to verify that claim with Philippe, empirically it seems correct in my test and there were comments related to that in the ROOT discussion of the bug (I overlooked those at first).

Reply:
I'll add the comment.

Reply:
> The original bug in ROOT in CMSSW_13_0_0 that started all this only affected non split classes.

Ok, thanks. Could you add this detail in the comment as well?